ratel/src/astgen/expression.rs

```diff
@@ -213,7 +213,23 @@ impl<'ast> SerializeInLoc for ArrayExpression<'ast> {
         S: Serializer,
     {
         self.in_loc(serializer, "ArrayExpression", 1, |state| {
-            state.serialize_field("elements", &self.body)
+
+            let mut elems = self.body.iter()
+                                     .map(|loc| if loc.item != Expression::Void { Some(loc) } else { None } )
+                                     .collect::<Vec<_>>();
+            loop {
+                let should_pop = match elems.last() {
+                    Some(&None) => true,
+                    Some(&Some(_)) => break,
+                    None => break,
+                };
+
+                if should_pop {
+                    elems.pop();
+                }
+            }
+
+            state.serialize_field("elements", &elems)
```

Comment:
There should be a way of doing this without having to reallocate elements on a `Vec`